src/Sarif/Writers/SarifLogger.cs

```diff
@@ -462,6 +463,28 @@ public void AnalyzingTarget(IAnalysisContext context)
         {
         }
 
+        public void TargetAnalysisCompleted(IAnalysisContext context)
+        {
+        }
+
+        public void LogMemoryUsage(IAnalysisContext context)
+        {
+            using (var currentProcess = Process.GetCurrentProcess())
+            {
+                LogToolNotification(
```

Comment:
>LogToolNotification [](http://example.com/codeflow?start=16&length=19)

you have implemented duplicate output in two different loggers. this isn't right. instead, you generate a notification upstream in the aggregating logger, which will be passed to the sarif logger and console logger, as every notification is.